src/main/java/org/example/LessonDao.java

```diff
@@ -89,31 +77,28 @@ public Lesson getLessonById(int lessonId) {
 
         try (Connection connection = DataBaseConnection.getConnection();
              PreparedStatement statement = connection.prepareStatement(query)) {
+            int lessonIDIndex = 1;
+            statement.setInt(lessonIDIndex, lessonId);
+            ResultSet resultSet = statement.executeQuery();
 
-            statement.setInt(1, lessonId);
-
-            try (ResultSet resultSet = statement.executeQuery()) {
-                if (resultSet.next()) {
-                    int lessonIdIndex = 1;
-                    int lessonNameIndex = 2;
-                    int homeworkIdIndex = 3;
-                    int homeworkNameIndex = 4;
-                    int homeworkDescriptionIndex = 5;
-
-                    int lessonIdResult = resultSet.getInt(lessonIdIndex);
-                    String lessonName = resultSet.getString(lessonNameIndex);
-                    int homeworkId = resultSet.getInt(homeworkIdIndex);
-                    String homeworkName = resultSet.getString(homeworkNameIndex);
-                    String homeworkDescription = resultSet.getString(homeworkDescriptionIndex);
-
-                    Homework homework = new Homework(homeworkId, homeworkName, homeworkDescription);
-                    lesson = new Lesson(lessonIdResult, lessonName, homework);
-                }
+            if (resultSet.next()) {
+                int index = 1;
+
+                int lessonIdResult = resultSet.getInt(index++);
+                String lessonName = resultSet.getString(index++);
+                int homeworkId = resultSet.getInt(index++);
+                String homeworkName = resultSet.getString(index++);
+                String homeworkDescription = resultSet.getString(index++);
```

Comment:
No need to increment `index` here, because it is the last column, and the next incremented value won't ever be used.